src/operator/tensor/elemwise_scatter_op.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "./elemwise_binary_op-inl.h"
+#include "./elemwise_binary_scalar_op.h"
+#include "./elemwise_scatter_op.h"
+
+namespace mxnet {
+namespace op {
+
+static bool StorageTypeRspOrDenseOutput(const NodeAttrs& attrs,
+                                        const int dev_mask,
+                                        DispatchMode* dispatch_mode,
+                                        std::vector<int>* in_attrs,
+                                        std::vector<int>* out_attrs) {
+  CHECK_EQ(in_attrs->size(), 2U);
+  CHECK_EQ(out_attrs->size(), 1U);
+  const NDArrayStorageType lhs_stype = static_cast<NDArrayStorageType>((*in_attrs)[0]);
+  if (lhs_stype == kRowSparseStorage) {
+    return storage_type_assign(&out_attrs[0], kRowSparseStorage,
+                               dispatch_mode, DispatchMode::kFComputeEx);
+  }
+  return storage_type_assign(&out_attrs[0], kDefaultStorage,
+                             dispatch_mode, DispatchMode::kFCompute);
+}
+
+static bool StorageTypeScatteredScalarOp(const NodeAttrs& attrs,
+                                        const int dev_mask,
+                                        DispatchMode* dispatch_mode,
+                                        std::vector<int>* in_attrs,
+                                        std::vector<int>* out_attrs) {
+  // Supports kDefaultStorage, kRowSparseStorage and kCSRStorage
+  const NDArrayStorageType stype = static_cast<NDArrayStorageType>((*in_attrs)[0]);
+  return storage_type_assign(out_attrs,
```

Comment:
Same here: do not return directly on storage_type_assign.. (e.g nd.scatter_plus(rsp,out = csr))

Reply:
Done